extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/DotNames.java

```diff
@@ -0,0 +1,26 @@
+package io.quarkus.arc.deployment.configproperties;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.jboss.jandex.DotName;
+
+import io.quarkus.arc.runtime.config.ConfigProperties;
+
+final class DotNames {
+
+    private DotNames() {
+    }
+
+    static final DotName OBJECT = DotName.createSimple(Object.class.getName());
+    static final DotName STRING = DotName.createSimple(String.class.getName());
+    static final DotName OPTIONAL = DotName.createSimple(Optional.class.getName());
+    static final DotName LIST = DotName.createSimple(List.class.getName());
+    static final DotName SET = DotName.createSimple(Set.class.getName());
```

Comment:
We don't support maps, here? (not that it's strictly necessary to get this PR in, it can be a later addition)

Reply:
Nope. I didn't see anything in MP Config for maps, but I could very well have missed it. 
A very good addition for the next iteration :)

Reply:
Yeah, I think the maps are an extension specifically developed by @dmlloyd .

Reply:
I'll check and see if there is a converter. If so, we can use it